crates/primitives/starknet/src/crypto/commitment/mod.rs

```diff
@@ -189,6 +192,7 @@ fn calculate_transaction_hash_common<T>(
     nonce: U256,
     version: u8,
     tx_prefix: &[u8],
+    chain_id: &str,
```

Comment:
Since I guess we can't fetch the chain_id from the GenesisConfig here, I'd to add this as function parameter. But this would cause hardcoding the chain_id at some places like when casting `DeclareTransaction` to `Transaction`. 
https://github.com/keep-starknet-strange/madara/pull/383/files#diff-d1e7254b6a1f394f0f8f9fee94b5cfad347b50e492ef7a6d92e7238220852faeR340
How can I fix this?